dotCMS/src/integration-test/java/com/dotmarketing/db/DBPropertiesDataSourceStrategyTest.java

```diff
@@ -0,0 +1,144 @@
+import com.dotmarketing.util.Constants;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.util.Date;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Tests for {@link DBPropertiesDataSourceStrategy} class
+ *
+ * @author nollymar
+ */
+public class DBPropertiesDataSourceStrategyTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void testExistsDBPropertiesFileShouldReturnFalse() {
+        final DBPropertiesDataSourceStrategy strategy = new DBPropertiesDataSourceStrategy(
+                new File("fake_file.properties"));
+        assertFalse(strategy.existsDBPropertiesFile());
+    }
+
+    @Test
+    public void testExistsDBPropertiesFileShouldReturnTrue() throws IOException {
+        final Path path = Files.createTempFile("test", String.valueOf(System.currentTimeMillis()));
+        final DBPropertiesDataSourceStrategy strategy = new DBPropertiesDataSourceStrategy(
+                path.toFile());
+        assertTrue(strategy.existsDBPropertiesFile());
+    }
+
+    @Test(expected = DotRuntimeException.class)
+    public void testApplyWithANonExistingFileShouldFail() {
+        new DBPropertiesDataSourceStrategy(new File("fake_file.properties")).apply();
+    }
+
+    @Test(expected = DotRuntimeException.class)
+    public void testApplyWithANullFileShouldFail() {
+        new DBPropertiesDataSourceStrategy(null).apply();
+    }
+
+    @Test
+    public void testApplyWithValidFileShouldPass()
+            throws IOException, SQLException, NamingException {
+        final HikariDataSource testDatasource = (HikariDataSource) TestInitialContext.getInstance()
+                .getDataSource();
+
+        final File tempFile = createTempFile(
+                "connection_db_driver=" + testDatasource.getDriverClassName() + "\n"
+                        + "connection_db_base_url=" + testDatasource.getJdbcUrl() + "\n"
+                        + "connection_db_username=" + testDatasource.getUsername() + "\n"
+                        + "connection_db_password=" + testDatasource.getPassword() + "\n"
+                        + "connection_db_validation_query=SELECT 1");
+
+        final DataSource dataSource = new DBPropertiesDataSourceStrategy(tempFile).apply();
+
+        assertNotNull(dataSource);
+        assertTrue(dataSource instanceof HikariDataSource);
+        assertNotNull(dataSource.getConnection());
+    }
+
+    @Test
+    public void testGetHikariConfigWithValidFileShouldPass()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)